novem/vis/grid.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import annotations
+
+from typing import Any
+
+from novem.vis import NovemVisAPI
+
+
+class Grid(NovemVisAPI):
+    """
+    Novem grid class
+    """
+
+    def __init__(self, id: str, **kwargs: Any) -> None:
+        """
+        :id grid name, duplicate entry will update the grid
+
+        """
+
+        # if we have an @ name we will override id and user
+        if id[0] == "@":
+            cand = id[1:].split("~")
+            id = cand[1]
+            uname = cand[0]
+            kwargs["user"] = uname
+
+        self.id = id
+
+        self._vispath = "grids"
+        self._type = "grid"
+
+        super().__init__(**kwargs)
+
+        self._parse_kwargs(**kwargs)
+
+    def __call__(self, content: Any, **kwargs: Any) -> Any:
+        """
+        Set's the layout of the grid
+
+        This paramter is expected to be one of the valid layout
+        strings or formats
+        """
+
+        raw_str: str = str(content)
+
+        # setting the content object will invoke a server write
+        self.layout = raw_str
+
+        # also update our varibales
+        self._parse_kwargs(**kwargs)
+
+        # return the original object so users can chain the contentframe
+        return content
+
+    def _parse_kwargs(self, **kwargs: Any) -> None:
+
+        # first let our super do it's thing
+        super()._parse_kwargs(**kwargs)
+
+        # get a list of valid properties
+        # exclude mapping and layout as they needs to be run last
+        props = [x for x in dir(self) if x[0] != "_" and x not in ["layout", "mapping", "read", "delete", "write"]]
+
+        do_layout = False
+        do_mapping = False
+        for k, v in kwargs.items():
```

Comment:
we're trying to get away from kwargs, not add more!

Reply:
This is just a convenient way to ensure that all `@property` setters are also available as `kwargs` on the `Class` init.

The reason for this is to ensure that both interactive use of novem like this:
```python
plt = Plot('test_plot')
plt.type = "gbar"
plt.title = "Simple test bar"
plt.data = df
plt.x
```

can also be done inline like this
```python
df.pipe(Plot("test_plot", type="gbar", title="Simple test bar")).x
```

Perhaps there is a better way of doing this, but with the current version adding a `@property` also introduces automatic support for `kwarg` parsing. So it's somewhat DRY.

This was a copy of existing code, so it is at least consistent. I encourage a possible rewrite and better documentation of this behaviour, but I'm not going to do it for this PR